internal/bootstrapping/agent.go

```diff
@@ -357,27 +357,29 @@ func (a *Agent) HandleResponse(inboxMsg *message.Message) ([]*message.Message, e
 	return nil, err
 }
 
-func (a *Agent) handleResponseMessage(msg *message.Message) (string, *MessageError, error) {
+func (a *Agent) handleResponseMessage(msg *message.Message) (string, *MessageError, bool, error) {
 	var rspMsg *protocol.Envelope
 	if err := json.Unmarshal(msg.Payload, &rspMsg); err != nil {
 		a.Logger.Error("Unexpected message format", err, watermill.LogFields{
 			"uuid": msg.UUID,
 		})
-		return "", NewMessageParameterInvalidError(), err
+		return "", NewMessageParameterInvalidError(), false, err
 	}
 
 	correlationID := rspMsg.Headers.CorrelationID()
+	respReq := rspMsg.Headers.IsResponseRequired()
+
 	var replyToID string
```

Comment:
It seems that this variable is not needed any more. The correlation identifier can be provided.